projects/core/src/routing/facade/routing-params.service.ts

```diff
@@ -0,0 +1,43 @@
+import { Injectable } from '@angular/core';
+import { ActivatedRouteSnapshot, NavigationEnd, Router } from '@angular/router';
+import { Observable } from 'rxjs';
+import { filter, map, shareReplay, startWith } from 'rxjs/operators';
+
+/**
+ * Service to expose all parameters for the router, including child routes.
+ * This is convenient in case the parent route (component) requires awareness
+ * of child routes parameters.
+ */
+@Injectable({ providedIn: 'root' })
+export class RoutingParamsService {
+  protected routerEvent$ = this.router.events.pipe(
+    filter((event) => event instanceof NavigationEnd),
+    startWith(this.router)
```

Comment:
1 `startWith(<something_that_is_not_event>)` should not be placed in `routerEvent$` observable - let's move it to the start of `params$` observable (to the line 21)
2. We don't make any use of the value `this.router` later on. What do you think about `startWith(undefined)` + comment why we do it:
   ```
   startWith(undefined), // emit value for consumer who subscribed after NavigationEnd event
   ```

Bagatelle: to be more precise, what do you think about renaming `routerEvent$` to `navigationEndEvent$`.

Reply:
Good feedback, thanks